### PR TITLE
F-16CM migration to module

### PR DESCRIPTION
**Pull request by jdahlblom**
There is something that is differing between old ```parse_indication``` and new ```Module.parse_indication```.
The module doesn't work with the latter.

### CODE REVIEW

**Comment by jdahlblom** [comment]
```
OLD : parse_indication(6)
Active Mode 3 Code : 0000
WPT IncDecSymbol : a
TACAN Label : T
Selected UHF Frequency : 305.00
VHF Label_placeholder : 
System Time :  0:00:00
VHF Label : VHF
Selected VHF Frequency : 127.00
Selected Steerpoint :   0
UHF Mode Rotary : UHF
UHF Mode Rotary_placeholder : 
DED CNI IFF PH : 
TACAN Band : X
TACAN Channel :   1
IFF Modes Label : M
DED CNI TACAN PH : 
IFF Modes Enabled :   
```

```
New : Module.parse_indication(6)
1 : 
2 : UHF
3 : 305.00
4 : 
5 : VHF
6 : 127.00
7 : 
8 : M
9 :       
10 : 0000
11 :   0
12 : a
13 :  0:00:00
14 : 
15 : T
16 :   1
Active Mode 3 Code : 0000
WPT IncDecSymbol : a
17 : X
TACAN Label : T
Selected UHF Frequency : 305.00
VHF Label_placeholder : 
System Time :  0:00:00
VHF Label : VHF
Selected VHF Frequency : 127.00
Selected Steerpoint :   0
UHF Mode Rotary : UHF
0 : 17
IFF Modes Enabled :       
DED CNI IFF PH : 
TACAN Band : X
TACAN Channel :   1
IFF Modes Label : M
DED CNI TACAN PH : 
UHF Mode Rotary_placeholder : 
```

```
Insight :

-----------------------------------------
UHF Mode Rotary_placeholder

children are {
-----------------------------------------
UHF Mode Rotary
UHF
}
-----------------------------------------
Selected UHF Frequency
305.00
-----------------------------------------
VHF Label_placeholder

children are {
-----------------------------------------
VHF Label
VHF
}
-----------------------------------------
Selected VHF Frequency
127.00
-----------------------------------------
DED CNI IFF PH

children are {
-----------------------------------------
IFF Modes Label
M
-----------------------------------------
IFF Modes Enabled
   4  
-----------------------------------------
Active Mode 3 Code
1337
}
-----------------------------------------
Steerpoint Use
STPT
-----------------------------------------
Selected Steerpoint
  1
-----------------------------------------
WPT IncDecSymbol
a
-----------------------------------------
System Time
 8:00:16
-----------------------------------------
DED CNI TACAN PH

children are {
-----------------------------------------
TACAN Label
T
-----------------------------------------
TACAN Channel
  1
-----------------------------------------
TACAN Band
X
}
```

I added a new function to logger and it might be incorrect but both ```parse_indications``` are logged with the same function.

**Comment by charliefoxtwo** [comment]
The new parse_indication adds some additional functionality for the KA-50 and F-14 but shouldn't have any info missing. Looking at what you logged, it looks correct and I can't see what would be different that would cause issues with this module 🤔

Which indications aren't working? Is it just some of them or all of them?

**Comment by jdahlblom** [comment]
> The new parse_indication adds some additional functionality for the KA-50 and F-14 but shouldn't have any info missing. Looking at what you logged, it looks correct and I can't see what would be different that would cause issues with this module 🤔
> 
> Which indications aren't working? Is it just some of them or all of them?

It crashed at this indication. I will look more today. Could be something else too now that I think of it. That DED code isn't really straightforward for me.

**Comment by WarLord211** [comment]
i have become an updated lua for the DED. How do we intergrade this?

**Comment by jdahlblom** [comment]
> i have become an updated lua for the DED. How do we intergrade this?

Discord?

**Comment by jdahlblom** [comment]
@charliefoxtwo you will have to look at ```Module.parse_indication()```. When using the old version it works, added radios and they work fine. So the only thing left is ```Module.parse_indication()```.

File: Scripts/DCS-BIOS/lib/modules/aircraft_modules/F-16C_50.lua [hunk 1]
```diff
@@ -0,0 +1,2016 @@
...
+
+	while intval > 0 do
+		retBytes[i] = intval % 256
+		intval = (intval - retBytes[i]) / 256
+		i = i + 1
+	end
+	return string.char(retBytes[1], retBytes[2], retBytes[3], retBytes[4])
+end
+
+OR, XOR, AND = 1, 3, 4
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/modules/aircraft_modules/F-16C_50.lua:1713]: These should probably be local
```suggestion
local OR, XOR, AND = 1, 3, 4
```

File: Scripts/DCS-BIOS/lib/modules/aircraft_modules/F-16C_50.lua [hunk 2]
```diff
@@ -0,0 +1,2016 @@
...
+		if curr_base ~= " " then
+			merged[i] = curr_base
+		else
+			merged[i] = string.sub(new_data, i, i)
+		end
+	end
+	return before .. table.concat(merged) .. after
+end
+
+function IntToByteString(intval)
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/modules/aircraft_modules/F-16C_50.lua:1698]:
```suggestion
local function IntToByteString(intval)
```

**Review by charliefoxtwo — COMMENTED** [review]
A couple of globals that cna be made local here

**Comment by jdahlblom** [comment]
```
2023-10-22 16:57:41.085 ERROR   Lua::Config (Main): Call error LuaExportAfterNextFrame:[string "C:\Users\user\Saved Games\DCS\Scripts\DCS-BIOS\lib\modules\aircraft_modules\F-16C_50.lua"]:1840: attempt to index local 'label' (a number value)
stack traceback:
	[C]: ?
	[string "C:\Users\user\Saved Games\DCS\Scripts\DCS-BIOS\lib\modules\aircraft_modules\F-16C_50.lua"]:1840: in function 'buildDEDLine'
	[string "C:\Users\user\Saved Games\DCS\Scripts\DCS-BIOS\lib\modules\aircraft_modules\F-16C_50.lua"]:1893: in function 'hook'
	[string "C:\Users\user\Saved Games\DCS\Scripts\DCS-BIOS\lib\BIOSStateMachine.lua"]:63: in function 'queue_module_data'
	[string "C:\Users\user\Saved Games\DCS\Scripts\DCS-BIOS\lib\BIOSStateMachine.lua"]:141: in function 'step'
	[string "C:\Users\user\Saved Games\DCS\Scripts\DCS-BIOS\BIOS.lua"]:227: in function 'LuaExportAfterNextFrame'
	[string "C:\Users\user\Saved Games\DCS\Scripts\DCS-INSIGHT\lib\DCSInsight.lua"]:89: in function <[string "C:\Users\user\Saved Games\DCS\Scripts\DCS-INSIGHT\lib\DCSInsight.lua"]:75>.
```

**Pull request merged**